FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
```

Comment:
I love regexes, but they can sometimes be arcane spells that you place in your program, and people have to just trust that they work. How would you go about testing that this regex works as intended?

Reply:
Testing:  I'd probably go the test script route

Another way: Instead of testing the input OnChange, I'd first change it so that it's validated after the user has clicked off the field. To get rid of the regex, we could create a function that loops through a char array of invalid characters with something like:

(using C# syntax)

foreach(char c in badCharArray)
{
if(emailString.Contains(c))
     return invalid;
}
return valid;


It would make it easier to read and modify if needed.

Reply:
> test script route
Could you elaborate on that? I'm talking, what are your test cases/strategy for testing something like this. 

I guess that there are 2 things you're concerned about here, keeping out "bad" characters and making sure the email matches the format. I think the regex is ok, but the problem with email regexes is that you'll always be able to find a "better" one. A decent one is fine as long as it fits the reasonable use cases.